gapic/src/main/com/google/gapid/views/DeviceValidationView.java

```diff
@@ -230,16 +231,35 @@ private void displayResult(DeviceValidationResult result) {
     notifyListeners(SWT.Modify, new Event());
   }
 
-  private String convertToMessage(ErrorCode errorCode) {
+  private String convertToMessage(ErrorCode errorCode, ValidatorType validatorType) {
+    String validatorName = "";
+    switch (validatorType) {
+      case Invalid:
```

Comment:
Agreed that the string builder situation is not ideal, but the system has to also support cached validation results which does not include info on the validator type.  With `return "Device successfully validated with " + validatorName + " validator."` it will not work correctly for validation results that were cached looking like `Device successfully validated with validator.`

It's the most read-able way to support both having no validator type or having a specific validator type. I think the appropriate solution here would be to include validator type in the cached result but it might be a little out-of-scope for this PR (i.e. making sure the cached results are still read & written properly between updates)

Reply:
Added TODO to clean this up along with filing cleanup bug